### PR TITLE
Docusaurus v3 upgrade

### LINKED ISSUE #212: Upgrade to docusaurus v3
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

As titled:

- https://github.com/facebook/docusaurus/releases/tag/v3.0.0
- docs: https://docusaurus.io/docs/migration/v3

### Motivation

We need to test the redirects mainly.

### Example

-

### PR DESCRIPTION
**Pull request by james-kaguru**
## Description

This upgraded the project dependancies to match up to what is required in docusaurus v3

## Related Issues

#212

## Check List

- [X ] I have read the [Contributing Guidelines](./CONTRIBUTING.md)

### CODE REVIEW

**Comment by james-kaguru** [comment]
@Eomm There are some issues causes by upgrading docusaurus to v3 noteable the formatting from some files in the fastify/fastify docs. The previous versions of the docs have formarting (notably ones identified by running npx docusaurus-mdx-checker). The updrage to v3 might have to include seperate docs for v2 and below and v3 and above. Not sure how to achieve this but I just wanted to raise a potential bottleneck. In addition orama plugin for docusaurua moved from @orama/plugin-docusaurus to @orama/plugin-docusaurus-v3 . I have temporarily commented out the plugin in docusaurus config because it brought up some issues but Im currently working to resolve it.

File: versions-shipped.json [hunk 1]
```diff
@@ -1 +1,187 @@
-[]
+[
```
Comment (Eomm) [inline versions-shipped.json]: the `version*` files should not be committed

**Comment by Eomm** [comment]
> There are some issues causes by upgrading docusaurus to v3 noteable the formatting from some files in the fastify/fastify docs

Some example?

I don't get if it breaks the website build or the linting.
In the latter case, no problem if we must upgrade the linting in this repo

**Comment by james-kaguru** [comment]
The docs from the previous versions have errors that have to be solved so that they can be run by docusaurus v3. I recently updated the docs in fastify/fastify to match the docusaurus requirements for v3 but since the docs prior to that were not correctly formated there is an error when i run npm start dev as shown below.
Screenshot 2024-05-20 at 9 53 21 PM
They are minor errors but Im not sure how we are going to update the previous docs and have the same version numbers

File: versions-shipped.json (hunk 1 above)
Comment (james-kaguru) [reply versions-shipped.json]: well noted

**Comment by Eomm** [comment]
> They are minor errors but Im not sure how we are going to update the previous docs and have the same version numbers

Can we do some regexp magic like this?

https://github.com/fastify/website/blob/e3e433a2f50abb2020a16bdf2a5754181d80b11d/scripts/process-releases.js#L105-L113

**Comment by james-kaguru** [comment]
Really smart approach...Lemmi try to come up with some magic to address the issue. Also Ive just remembered that to ensure the docs in fastify/fastify continue to be compatible with docusaurus v3 we also need to ensure it runs 
```
npx docusaurus-mdx-checker
```
as a requirement during build time

**Comment by Eomm** [comment]
Need to check this PR - I will do it - I'm just doing to many things 🥲

**Comment by james-kaguru** [comment]
@Eomm I also got a bit held up but Im back on it to ensure we have passing build before you review it..there some hitches as I was trying to upgrade but will notify you incase I make a breakthrough

**Comment by Eomm** [comment]
Sorry @james-kaguru , I want to ship this PR very soon 😮‍💨

**Review by Eomm — APPROVED** [review]
FINALLY!

Thanks @james-kaguru

**Pull request merged**